.github/workflows/reuse.yml

```diff
@@ -0,0 +1,254 @@
+        type: string
+        required: false
+        default: "balena/testbot-rig"
+      BALENACLOUD_API_URL:
+        type: string
+        required: false
+        default: "balena-cloud.com"
+      BALENAMACHINE_API_URL:
+        required: false
+        type: string
+        default: "bm.balena-dev.com"
+      BALENACLOUD_SSH_URL:
+        required: false
+        type: string
+        default: "ssh.devices.bm.balena-dev.com"
+      BALENACLOUD_SSH_PORT:
+        default: 222
+        type: number
+        required: false
+      QEMU_CPUS:
+        required: false
+        type: number
+        default: 4
+      QEMU_MEMORY:
+        type: string
+        required: false
+        default: "1G"
+      WORKER_TYPE:
+        description: "Valid choices: testbot/qemu"
+        type: string
+        required: true
+      LEVIATHAN_DEBUG:
+        description: |
+          Use this parameter to modify the default debugging settings on test runs. To debug test runs when rebuilding, add new values to this parameter separated with commas. Know more about the possible values available for debugging: https://balena-os.github.io/leviathan/pages/Getting-Started/debugging.html Example value: dev: true, failFast: false
+        type: string
+        required: false
+    secrets:
+      BALENACLOUD_API_KEY:
+        required: false
+      BALENAMACHINE_API_KEY:
+        required: false
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Get Leviathan checked out
+      uses: actions/checkout@v4
+      with:
+        repository: balena-os/leviathan
+        submodules: true
+        # path: ./leviathan-workspace
+        # The branch, tag or SHA to checkout. When checking out the repository that
+        # triggered a workflow, this defaults to the reference or SHA for that event.
+        # Otherwise, uses the default branch.
+        ref: ${{ inputs.LEVIATHAN_REF }}
+
+    # - name: Set up Docker Buildx
+    #   uses: docker/setup-buildx-action@v3
+    # Don't think, I need this since docker-compose uses only buildkit which we enabled by default in Makefile
+
+    - name: Build it, and they might come
+      shell: bash
+      env:
+        WORKER_TYPE: ${{ inputs.WORKER_TYPE }}
+        # SECUREBOOT: ${{ inputs.SECUREBOOT }}
+      run: |
+        make config
+        make build
+        docker image ls -a
+
+    - name: Save the images
+      shell: bash
+      run: |
+        docker save --output /tmp/corelatest.tar leviathan_core:latest 
+        docker save --output /tmp/clientlatest.tar leviathan_client:latest 
+
+    - name: Upload Core artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: corelatest.tar
+        path: /tmp/corelatest.tar
+        compression-level: 7
+
+
+    - name: Upload Client artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: clientlatest.tar
+        path: /tmp/clientlatest.tar
+        compression-level: 7
```

Comment:
I think unless the test step is a matrix, which I don't think it ever will be, we don't actually gain anything by separating the build and the test into jobs.

We just lose time on uploading and downloading the artifacts.

Reply:
Why can't the test step be a matrix? I was hoping for it to be definitely a matrix using the same build images for every test run of SUITE + ENV + WORKER_TYPE

Reply:
If we can achieve a matrix of tests at least what I am thinking, we can truly achieve the quick test suite specific retries for that sweet sweet feedback on PRs.

Reply:
personally I would prefer the test step be a matrix, for these reasons.  Even if that is a future improvement and out of scope for this cycle (there are probably implications we haven't considered), it opens the route to doing that

Reply:
yeah, in a reusable workflow for self-tests it makes sense to have a matrix

we still haven't decided if we need this workflow as it duplicates a lot of logic already in the yocto workflow